controllers/clustergroupupgrade_controller.go

```diff
@@ -281,24 +266,189 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 	return nextReconcile, nil
 }
 
+func (r *ClusterGroupUpgradeReconciler) initializeRemediationPolicyForBatch(
+	clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) {
+
+	if clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt.IsZero() {
+		clusterGroupUpgrade.Status.Status.RemediationPlanForBatch = make(map[string]int)
+		batchIndex := clusterGroupUpgrade.Status.Status.CurrentBatch - 1
+
+		// By default, assume all clusters start applying all policies from the first policy.
```

Comment:
Yes, I had this approach initially, where I started with the first policy, but we all discussed (sorry, you were away on vacation) that it takes quite a lot of computing to determine which is the first managed policy in which the cluster is non compliant (we need to go through all the policies and check all the NonCompliant clusters...), so we decided to just go through all the policies from the start.

Reply:
Sounds good. Do you know how long it takes to step through each policy? Is each iteration on the order of seconds or minutes or ???

Reply:
We decide this by setting the nextReconcile variable. If we reconcile after an hour, then it's not good because we're basically waiting for an hour for a policy that we don't actually apply. I'm actually debating myself that it might be better to only include the next policy for which the policy is NonCompliant. This way we are not wasting reconciling iterations on policies for which the cluster is compliant... We do use a bit more time to get that info up front, but we save from the total upgrade time. Yep.

Reply:
Decided to consider only the NonCompliant policies to save some upgrade time. :crossed_fingers: